contrib/drivers/lvm/targets/iscsi.go

```diff
@@ -16,69 +16,43 @@ package targets
 
 import (
 	"fmt"
+	"os"
 	"os/exec"
 	"strconv"
 	"strings"
 
 	log "github.com/golang/glog"
+	"github.com/opensds/opensds/pkg/utils"
 )
 
-type ISCSITarget interface {
-	CreateISCSITarget() error
-	GetISCSITarget() int
-	RemoveISCSITarget() error
+const (
+	opensdsPrefix = "opensds-"
+	tgtAdminCmd   = "tgt-admin"
+)
 
-	AddLun(lun int, path string) error
+type ISCSITarget interface {
+	CreateISCSITarget(volId, tgtIqn, path, initiator string, chapAuth []string) error
+	GetISCSITarget(iqn string) int
+	RemoveISCSITarget(volId, iqn string) error
 	GetLun(path string) int
-	RemoveLun(lun int) error
-
-	BindInitiatorName(initiator string) error
-	UnbindInitiatorName(initiator string) error
-
-	BindInitiatorAddress(initiator string) error
-	UnbindInitiatorAddress(initiator string) error
 }
 
-func NewISCSITarget(tid int, name string, bip string) ISCSITarget {
+func NewISCSITarget(bip, tgtConfDir string) ISCSITarget {
 	return &tgtTarget{
-		Tid:    tid,
-		TName:  name,
-		BindIp: bip,
+		TgtConfDir: tgtConfDir,
+		BindIp:     bip,
 	}
 }
 
 type tgtTarget struct {
-	Tid    int
-	TName  string
-	BindIp string
-}
-
-func (t *tgtTarget) AddLun(lun int, path string) error {
-	var cmd = []string{
-		"--lld", "iscsi",
-		"--op", "new",
-		"--mode", "logicalunit",
-		"--tid", fmt.Sprint(t.Tid),
-		"--lun", fmt.Sprint(lun),
-		"--backing-store", path,
-	}
-	if _, err := t.execCmd(cmd); err != nil {
-		log.Error("Fail to exec 'tgtadm' to add lun into iscsi target:", err)
-		return err
-	}
-
-	return nil
+	BindIp     string
+	TgtConfDir string
 }
 
 func (t *tgtTarget) GetLun(path string) int {
```

Comment:
Is there special use case for keeping `GetLun` method?

Reply:
The ConnectionData  need this id.

Reply:
Got it